AppController/lib/taskqueue.rb

```diff
@@ -168,21 +167,25 @@ def self.start_slave(master_ip, clear_data, verbose)
     Djinn.log_debug('Waiting for RabbitMQ on master node to come up')
     HelperFunctions.sleep_until_port_is_open(master_ip, SERVER_PORT)
 
-    # we now reset it to join the head node. To do this we need the
-    # hostname of the master node. We go through few options:
-    # - the old one is to look into /etc/hosts for it
-    # - another one is to just try to resolve it
-    # - finally we give up and use the IP address
-    master_tq_host = `cat /etc/hosts | grep #{master_ip} | tr -s \" \" | cut -d \" \" -f2`.chomp
-    if master_tq_host.empty?
-      begin
-        master_tq_host = Resolv.getname(master_ip)
-      rescue Resolv::ResolvError
-        # We couldn't get the name: let's try to use the IP address.
-        master_tq_host = master_ip
+    # We need not to contact the master node, and we need to use its name
+    # (just the hostname not the fqdn).  To resolve it we use
+    # Addrinfo.getnameingo since Resolv will not use /etc/hosts and this
+    # could cause issues on private clusters.
+    master_tq_host = nil
+    begin
+      master_tq_host = Addrinfo.ip(master_ip).getnameinfo[0]
+      if (master_tq_host =~ /[[:digit:]]/).nil?
```

Comment:
I don't understand the regex here. Why does the fqdn need to have a digit?

Reply:
It doesn't: if the results starts with a number it is an IP address (since no DNS entry can start with a number) hence the warning about using an IP instead of a hostname (which is what rabbitmq wants).

Reply:
Your regex appears to be testing whether or not the fqdn has a digit anywhere in the string. For example,
```
irb(main):001:0> "ubuntu-bionic" =~ /[[:digit:]]/
=> nil
```
So if the fqdn does not contain a digit, it will log a warning and use the fqdn. If it does contain a digit, it will use just the hostname.

Reply:
Am I reading the if stmt wrong? My ruby-fu is weak...
```
if (master_tq_host =~ /^[[:digit:]]/).nil?
```
Doesn't that mean if the master_tq_host doesn't match the regex? eg the regex match is nil. So the if stmt would be true if its a hostname?

```
irb(main):008:0> if ("www.google.com" =~ /^[[:digit:]]/).nil?
irb(main):009:1>    puts "I'm a hostname"
irb(main):010:1> else
irb(main):011:1>    puts "I'm an ip address"
irb(main):012:1> end
I'm a hostname
```
I think the if/else blocks need to be reversed.

Actually maybe this is clearer:
```
# Warn if the hostname is an IP address
if /^[[:digit:]]{1}/.match?(master_tq_host)
    Djinn.log_warn("#{master_ip} didn't resolve to a hostname! Expect" \
                            " problems with rabbitmq clustering.")
else
    master_tq_host = master_tq_host.split('.')[0]
```

Reply:
ah yes! thanks you! Forgot to change the PR: the intention was to have something like unless (master_tq_host =~ /^[[:digit:]]/).nil? let me fix it.